osu.Game/Beatmaps/Formats/LegacyStoryboardDecoder.cs

```diff
@@ -341,6 +341,12 @@ private Anchor parseOrigin(string value)
             }
         }
 
+        private AnimationLoopType parseAnimationLoopType(string value)
+        {
+            var parsed = (AnimationLoopType)Enum.Parse(typeof(AnimationLoopType), value);
```

Comment:
Have you tried using `Enum.TryParse` here? does that avoid the need for the second check?

Reply:
It does not. Enum parsing methods will parse any string with an integer inside perfectly fine as long as it fits in the underlying storage type for the enum. It's a scarily wonky part of enums in general, actually.

[sharplab for reference.](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQcDMACR2DC2A3utmbjgKZQQC22AKpQM7DGnmcCCANB52QBC/bAF8RIxAnZoB5AKI1aAOgZgAngAUAhmGaUAPE1YA+ABQAiAKxWLPbAHsIbAG67sAB136AJgEoAbhFOOCQATjMvPUp/INlycTRRIA===)

Reply:
Aha, missed that detail. Makes sense.